### ISSUE TITLE
FEAUTURE: Inline Whiteboard

### ISSUE THREAD

**Issue by pylang** [jupyter/notebook#1999]
I do almost everything in the notebooks, but I wish I could make quick sketches and save them too, as I do in physical notebooks.  I think jupyter notebooks would be unlike any other if they supported handwritten sketches.  

I imagine something like a "canvas" cell type, that triggers a simple javascript whiteboard in place of the cell and accepts cursor inputs for sketching.  For example, a cell that was able to produce the following:

![2016-12-20 18_18_21-untitled - paint](https://cloud.githubusercontent.com/assets/10778668/21371789/4c12aeb2-c6e1-11e6-9560-61d176a9c2d4.png)

Based on this [SO Post](https://stackoverflow.com/questions/6845682/real-time-collaborative-drawing-whiteboard-in-html5-js-and-websockets), there appear to be ways to implement simple [JS-like whiteboards](http://www.unionplatform.com/?page_id=2762).  

Before TLDR; is this a feasible request in the next versions of the notebook/jupyterlab?  Has this feature has been reviewed before?  Thank you.

**Comment by rgbkrk** [comment]
That would be super neat! I've wanted this as well. On a Windows surface (pro|book), it would be nice to be able to pop open an area to start drawing in a similar fashion to how we do in OneNote.

UI wise, I'd love to be able to write directly into a new cell as a canvas area and save that as `image/png`. It's up to a notebook implementation to handle this as they see fit.

As for implementation, I think it would be nice if there was a generic yet specified field for mime bundles similar to `display_data` from outputs. Something like:

```
cell_type: "mimebundle",
data: {
  "image/png": ...
}
```

Clearly I'm spitballing here. I know I opened a related issue for being able to put `text/html` on the page without requiring magics (for the benefit of other kernels than IPython). Basically, some way to flip between edit and view mode for a singular mimetype.

**Comment by rgbkrk** [comment]
Related issue: https://github.com/jupyter/notebook/issues/1123

**Comment by pylang** [comment]
I second @rgbkrk's idea on saving to png.

**Comment by pylang** [comment]
Re: editing canvas-like images in notebooks
https://github.com/uclmr/egal

**Comment by sivark** [comment]
I came hear to suggest the same feature, to notice that a discussion is already underway! This feature would take jupyter-notebook further along in the direction of being a versatile "notebook".

**Comment by evertheylen** [comment]
I too would like to draw some sketches within the notebook. In my search, I've encountered [literallycanvas](http://literallycanvas.com/) which seems to be a perfect fit (various ways of drawing, adding text, JSON load/store, etc).

**Comment by n1000** [comment]
Not adding anything constructive to this discussion, just saying it would be awesome if this were added... 👍

**Comment by alxgom** [comment]
Just another person requesting for this. I have to teach at the uni and I think having a mix of plotting/interactivity, with latex capabilitys and the hability to do a hand written lecture would be perfect.

**Comment by Alchemication** [comment]
Brilliant idea. Came here to flag same!

**Comment by alkuzman** [comment]
+1

**Comment by NickAlger** [comment]
Yes +1. This would be awesome. The extension egal kinda does this, but it doesn't work very well

**Comment by srkunze** [comment]
+1

recently saw this https://twitter.com/tldraw/status/1461011977791234051

or this

https://github.com/excalidraw/excalidraw

**Comment by williamstein** [comment]
Of related interest, I recently implemented a Jupyter whiteboard for CoCalc:  https://doc.cocalc.com/whiteboard.html

It's a basic whiteboard and there's a Jupyter button off to the left that you can click on to create a new Jupyter code cell.  The result is that you get a whiteboard with Jupyter code cells in it, which is kind of fun to use as an alternative UI to Jupyter notebooks.  This is sorted of inverted from the question of this ticket though, which is having whiteboards in a notebook...

**Comment by mattveraldi** [comment]
Any news? I'd like that feature too

**Comment by jtpio** [comment]
This extension by @kolibril13 looks pretty good too: https://github.com/kolibril13/jupyter-tldraw

In any case, such functionality would probably still be implemented as an extension for the time being. So we could close this issue, and maybe link the currently available options in the top comment?

**Comment by jtpio** [comment]
Note the same issue in JupyterLab may have more context and pointers: https://github.com/jupyterlab/jupyterlab/issues/9194
